src/main/java/de/qabel/desktop/hockeyapp/HockeyAppClient.java

```diff
@@ -153,9 +166,8 @@ public HockeyAppVersion createNewVersion(String version) throws IOException {
         String responseContent = EntityUtils.toString(response.getEntity());
         JSONObject parsedJson = new JSONObject(responseContent);
 
-        if(getCurrentHockeyVersion() == null){
-            currentHockeyVersion = new HockeyAppVersion(parsedJson.getInt("id"),parsedJson.getString("shortversion"));
-        }
+        setCurrentHockeyVersion(new HockeyAppVersion(parsedJson.getInt("id"),parsedJson.getString("shortversion")));
```

Comment:
This doesn't deal with invalid json. If you ever get JSON where those 2 fields are not available the app will crash.